src/HDF5.jl

```diff
@@ -1329,28 +1329,18 @@ end
 
 # Read OPAQUE datasets and attributes
 function Base.read(obj::DatasetOrAttribute, ::Type{Opaque})
-    local buf
-    local len
-    local tag
-    sz = size(obj)
     obj_type = datatype(obj)
-    try
-        len = h5t_get_size(obj_type)
-        buf = Vector{UInt8}(undef,prod(sz)*len)
-        tag = h5t_get_tag(obj_type)
-        if obj isa Dataset
-            read_dataset(obj, obj_type, buf)
-        else
-            read_attribute(obj, obj_type, buf)
-        end
-    finally
-        close(obj_type)
-    end
-    data = Array{Array{UInt8}}(undef,sz)
-    for i = 1:prod(sz)
-        data[i] = buf[(i-1)*len+1:i*len]
+    sz  = size(obj)
+    buf = Matrix{UInt8}(undef, sizeof(obj_type), prod(sz))
+    if obj isa Dataset
+        read_dataset(obj, obj_type, buf, obj.xfer)
+    else
+        read_attribute(obj, obj_type, buf)
     end
-    Opaque(data, tag)
+    tag = h5t_get_tag(obj_type)
+    close(obj_type)
+    data = [buf[:,i] for (i, _) in enumerate(CartesianIndices(sz))]
```

Comment:
`for i in prod(sz)` ?

Reply:
`enumerate(CartesianIndices(sz))` has a shape while `prod(sz)` doesn't:

```julia
julia> [i for (i, _) in enumerate(CartesianIndices((2,2)))]
2×2 Matrix{Int64}:
 1  3
 2  4

julia> [i for i in 1:prod((2,2))]
4-element Vector{Int64}:
 1
 2
 3
 4
```

Reply:
I could make it `reshape([buf[:,i] for i in 1:prod(sz)], sz...)`, though.

Reply:
Oh my bad,  I was looking at the other PR too long and automatically assumed this should be `prod(sz)` without spending much time looking at the details.

Reply:
> I could make it reshape([buf[:,i] for i in 1:prod(sz)], sz...), though.

Perhaps one argument for this is that it matches `reshape` we do for `read`